### PR TITLE
sql: Add statement telemetry, and counters for savepoint stmts

### PR DESCRIPTION
**Pull request by bdarnell**
cc @awoods187

### CODE REVIEW

**Review by dt — APPROVED** [review]

**Review by knz — COMMENTED** [review]
The change LGTM, but this will need a release note to explain that the data showed in the admin UI is changing.

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35898#-:-LaSg98d7qXksV0aOcNf:b-6rrg69)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz)

**Review by knz — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 5 of 5 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35898)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by bdarnell** [comment]
@knz What data in the admin UI is changing? The only (intentional) change to existing data here is that savepoint commands are moved out of the "misc" category into their own counts. But we don't display the misc count or use it anywhere that I can see.

File: pkg/sql/exec_util.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ var (
 	}
 	MetaTxnAbort = metric.Metadata{
 		Name:        "sql.txn.abort.count",
-		Help:        "Number of SQL transaction ABORT statements",
+		Help:        "Number of SQL transaction abort errors",
```
Comment (knz) [inline pkg/sql/exec_util.go]: This is the user-visible change I was thinking about. Maybe it's innocuous?

Comment (bdarnell) [reply pkg/sql/exec_util.go]: Yeah, it's innocuous. The data isn't changing, I'm just fixing the help text to match the implementation (and is this help text even shown in the UI? I can't find it).

**Comment by bdarnell** [comment]
bors r=dt,knz

**Pull request merged**